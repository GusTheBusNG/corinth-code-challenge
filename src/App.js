import React, { useState } from 'react';
import './App.scss';
import Search from './components/search';
import Profile from './components/profile';
import { imageUrl } from './utils/cloudinary';

const App = () => {
  const [character, setCharacter] = useState({})

  return (
    <div className='content'>
      <div className='heading'>
        <img src={imageUrl('logo')} className='sw-logo' alt='Star Wars Logo' />
        <Search setCharacter={setCharacter} />
      </div>
      <Profile {...character} />
    </div>
  )
}

export default App;
