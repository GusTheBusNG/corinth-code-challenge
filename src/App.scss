@use 'assets/abstracts' as *;

html {
  background-image: url('https://res.cloudinary.com/dnglix0tu/image/upload/v1659232491/LifeWay-star-wars/background.jpg');
}

body {
  font-family: $family;
}

.content {
  display: flex;
  justify-content: center;
}

.heading {
  position: absolute;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.sw-logo {
  margin-top: 1rem;
}

h1, h5, p {
  margin: 0;
}