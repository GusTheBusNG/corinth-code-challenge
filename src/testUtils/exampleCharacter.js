export const character = {
  name: 'Anakin Skywalker',
  height: '6',
  mass: 'something',
  hair_color: 'brown',
  birth_year: 'some time long ago',
  species: ['example url species'],
  starships: ['example url starships'],
  films: ['example url fil1s']
}
