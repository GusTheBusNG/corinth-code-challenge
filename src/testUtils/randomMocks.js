export const stubLoading = (message = 'loading') => {
  const random = jest.requireActual('../utils/randomPhrases');
  random.getLoadingQuote = jest.fn(() => message)
}

export const stubError = (message = 'error') => {
  const random = jest.requireActual('../utils/randomPhrases');
  random.getErrorQuote = jest.fn(() => message)
}