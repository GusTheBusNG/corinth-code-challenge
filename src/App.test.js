import React from 'react';
import { act, create } from 'react-test-renderer';

import App from './App'
import Profile from './components/profile';
import Search from './components/search';
import { stubLoading } from './testUtils/randomMocks'
import { character } from './testUtils/exampleCharacter';

const createApp = () => create(<App />)

describe('App', () => {
  describe('render', () => {
    it('renders correctly', () => {
      stubLoading()
      expect(createApp().toJSON()).toMatchSnapshot();
    })
  })

  describe('when a user selects a character', () => {
    it('renders correctly', async () => {
      stubLoading()
      const app = createApp();
      const profile = await app.root.findByType(Profile);
      const search = await app.root.findByType(Search);

      act(() => search.props.setCharacter(character));

      expect(app.toJSON()).toMatchSnapshot();
      expect(profile.props.name).toEqual(character.name);
    })
  })
})
