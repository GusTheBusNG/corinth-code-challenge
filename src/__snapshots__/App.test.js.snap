// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render renders correctly 1`] = `
<div
  className="content"
>
  <div
    className="heading"
  >
    <img
      alt="Star Wars Logo"
      className="sw-logo"
      src="https://res.cloudinary.com/dnglix0tu/image/upload/f_auto/v1/LifeWay-star-wars/logo?_a=ATAMhAA0"
    />
    <div
      className="search"
    >
      <input
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="I'm looking for..."
        type="text"
        value=""
      />
      <div
        className="auto-complete"
      >
        <p
          className="loading"
        >
          loading
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`App when a user selects a character renders correctly 1`] = `
<div
  className="content"
>
  <div
    className="heading"
  >
    <img
      alt="Star Wars Logo"
      className="sw-logo"
      src="https://res.cloudinary.com/dnglix0tu/image/upload/f_auto/v1/LifeWay-star-wars/logo?_a=ATAMhAA0"
    />
    <div
      className="search"
    >
      <input
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="I'm looking for..."
        type="text"
        value=""
      />
      <div
        className="auto-complete"
      >
        <p
          className="loading"
        >
          loading
        </p>
      </div>
    </div>
  </div>
  <div
    className="profile"
  >
    <h1>
      Hello there 
      Anakin Skywalker
    </h1>
    <p
      className="bio"
    >
      6
      cm | 
      something
      kg | 
      some time long ago
       | 
      brown
    </p>
    <div
      className="info-pill"
    >
      <img
        alt="Species"
        className="icon"
        src="https://res.cloudinary.com/dnglix0tu/image/upload/f_auto/v1/LifeWay-star-wars/Species?_a=ATAMhAA0"
      />
      <p
        className="content"
      >
        loading
      </p>
    </div>
    <div
      className="info-pill"
    >
      <img
        alt="Starship"
        className="icon"
        src="https://res.cloudinary.com/dnglix0tu/image/upload/f_auto/v1/LifeWay-star-wars/Starship?_a=ATAMhAA0"
      />
      <p
        className="content"
      >
        loading
      </p>
    </div>
    <div
      className="films"
    >
      <img
        alt="film"
        className="film-poster"
        src="https://res.cloudinary.com/dnglix0tu/image/upload/f_auto/v1/LifeWay-star-wars/movies/1?_a=ATAMhAA0"
      />
    </div>
  </div>
</div>
`;
